test/windows/windows_shim.c

```diff
@@ -104,11 +97,24 @@ pthread_rwlock_unlock(pthread_rwlock_t *rwlock)
 	return (0);
 }
 
+int
+pthread_rwlock_tryrdlock(pthread_rwlock_t *rwlock)
+{
+	return (TryAcquireSRWLockShared(&rwlock->rwlock) ? 0 : EBUSY);
+}
+
+int
+pthread_rwlock_rdlock(pthread_rwlock_t *rwlock)
+{
+	AcquireSRWLockShared(&rwlock->rwlock);
+	return (0);
+}
+
 int
 pthread_rwlock_trywrlock(pthread_rwlock_t *rwlock)
 {
 	if (TryAcquireSRWLockExclusive(&rwlock->rwlock)) {
-		rwlock->exclusive_locked = GetCurrentThreadId();
+		rwlock->exclusive_locked = 1;
```

Comment:
Tracking the thread that has exclusive access seems like useful information: why did you take that out?

Reply:
We were mixing-and-matching `DWORD` and `int` types and we never used the value. If it's potentially interesting, I'll revert that part of the change and fix the type declaration instead.

Pushed in e1472a1.